bounded_counter/count_min_sketch.py

```diff
@@ -0,0 +1,99 @@
+import os
+import numpy
+import xxhash
+
+
+def hash_key(i, key):
+    return xxhash.xxh64(str(key), seed=i).intdigest()
+
+algorithm_basic = 0
+algorithm_conservative = 1
+algorithm_logcounter = 2
+
+class CountMinSketch(object):
+    def __init__(self, width=16, depth=8, algorithm = None):
+        self.logshift = 0
+        array_type = numpy.uint32
+        self.algorithm = algorithm_basic
+        self.inc_method = self.basic_increment
+        if algorithm and str(algorithm).lower() == 'conservative':
```

Comment:
`algorithm` should have a clear type (such as str, or unicode). No tests, conversions and postprocessing needed or desired. Python is a  duck-typed language.